src/adapter/src/subscribe.rs

```diff
@@ -49,6 +52,33 @@ pub struct ActiveSubscribe {
 }
 
 impl ActiveSubscribe {
+    pub(crate) fn initialize(&self) {
+        // Always emit progress message indicating snapshot timestamp.
+        self.send_progress_message(&Antichain::from_elem(self.as_of));
+    }
+
+    fn send_progress_message(&self, upper: &Antichain<Timestamp>) {
+        if self.emit_progress {
+            let upper = upper
+                .as_option()
+                .expect("upper must not be empty when sending a progress message");
```

Comment:
```suggestion
        if !self.emit_progress {
            return;
        }
        if let Some(upper) = upper.as_option() {
```

This was my idea.

Reply:
The option *could* be empty in the empty antichain case if the source has closed, so we do not want to panic in that case.

Reply:
done